packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.ts

```diff
@@ -1770,6 +1765,28 @@ async function getOptionArgsSuggestions(
         );
 
         if (option.name === 'by') {
+          // Add quick snippet for for stats ... by bucket(<>)
+          if (command.name === 'stats') {
+            suggestions.push({
+              label: i18n.translate(
+                'kbn-esql-validation-autocomplete.esql.autocomplete.addDateHistogram',
+                {
+                  defaultMessage: 'Add date histogram',
+                }
+              ),
+              text: ADD_DATE_HISTOGRAM_SNIPPET,
+              asSnippet: true,
+              kind: 'Function',
+              detail: i18n.translate(
+                'kbn-esql-validation-autocomplete.esql.autocomplete.addDateHistogramDetail',
+                {
+                  defaultMessage: 'Add date histogram using bucket()',
+                }
+              ),
+              sortText: '1A',
```

Comment:
I think we should open the suggestions menu when the user accepts the date histogram snippet

```suggestion
              sortText: '1A',
              command: TRIGGER_SUGGESTIONS_COMMAND,
```

Reply:
Updated here [`ca2d8e7` (#190828)](https://github.com/elastic/kibana/pull/190828/commits/ca2d8e773a2aabed6ffd1c3668339ea851d12d2a)